spec/lib/two_factor_authentication/models/two_factor_authenticatable_spec.rb

```diff
@@ -77,15 +77,22 @@
         @totp_helper = TotpHelper.new(instance.otp_secret_key, instance.class.otp_length)
       end
 
-      def do_invoke(code, user)
-        user.authenticate_totp(code)
+      def do_invoke(code, user, options = {})
+        user.authenticate_totp(code, options)
       end
 
       it 'authenticates a recently created code' do
         code = @totp_helper.totp_code
         expect(do_invoke(code, instance)).to eq(true)
       end
 
+      context 'when secret key passed in options' do
+        it 'authenticates a recently created code' do
+          code = @totp_helper.totp_code
+          expect(do_invoke(code, instance, {otp_secret_encryption_key: instance.otp_secret_encryption_key})).to eq(true)
+        end
+      end
+
```

Comment:
I feel like this spec is not very validating for the correct functionality. Shouldn't there be one spec which creates another instance, and passes its otp_secret_encryption_key in the options and makes sure the code from the first instance doesn't work, and another spec which shows that the code from the new instance works or something to that tune.